omod/src/test/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResourceTest.java

```diff
@@ -114,31 +137,39 @@ public void testCreateScheduledReport() throws Exception {
 	@Test
 	public void testCreateWithReportDefinitionAsParameters() throws Exception {
 		
+//		String reportRequestJson = "{\n" +
+//				"  \"status\": \"REQUESTED\",\n" +
+//				"  \"priority\": \"NORMAL\",\n" +
+//				"  \"reportDefinition\":" + REPORT_DEFINITION_JSON + "," +
+//				"  \"renderingMode\": \"org.openmrs.module.reporting.report.renderer.CsvReportRenderer\"\n" +
+//				"}";
+
 		String reportRequestJson = "{\n" +
 				"  \"status\": \"REQUESTED\",\n" +
 				"  \"priority\": \"NORMAL\",\n" +
-				"  \"reportDefinition\":" + REPORT_DEFINITION_JSON + "," +
-				"  \"renderingMode\": \"org.openmrs.module.reporting.report.renderer.CsvReportRenderer\"\n" +
+				"  \"reportDefinition\":" + REPORT_DEFINITION_JSON + ",\n" +
+				"  \"schedule\": \"0 56 11 ? * 3\",\n" +
+				"  \"renderingMode\":" + RENDERER_MODE_JSON +
 				"}";
 
 		SimpleObject properties = SimpleObject.parseJson(reportRequestJson);
 		RequestContext context = new RequestContext();
 		context.setRepresentation(Representation.DEFAULT);
 		SimpleObject response = (SimpleObject) getResource().create(properties, context);
 		assertNotNull(response.get("uuid"));
-		assertEquals(response.get("status"), ReportRequest.Status.REQUESTED);
+		assertEquals(response.get("status"), ReportRequest.Status.SCHEDULED);
 		assertEquals(response.get("priority"), ReportRequest.Priority.NORMAL);
 		SimpleObject resultObject = (SimpleObject) response.get("renderingMode");
 		String rendererType = (String) resultObject.get("rendererType");
 		assertEquals(rendererType, "org.openmrs.module.reporting.report.renderer.CsvReportRenderer");
 
 		ReportRequest request = getResource().getByUniqueId((String) response.get("uuid"));
 		assertThat(request.getReportDefinition().getParameterizable().getUuid(), is(REPORT_DEFINITION_UUID));
-		assertEquals(request.getReportDefinition().getParameterMappings().get("startDate"), "2017-01-01");
-		assertEquals(request.getReportDefinition().getParameterMappings().get("endDate"), "2017-01-31");
+		assertEquals(request.getReportDefinition().getParameterMappings().get("startDate").toString(), "Sun Jan 01 00:00:00 CET 2017");
+		assertEquals(request.getReportDefinition().getParameterMappings().get("endDate").toString(), "Tue Jan 31 00:00:00 CET 2017");
```

Comment:
@druchniewicz - I checked out this branch and built it as a sanity check prior to merging, and this unit test fails due to the dependency on this timezone.  That needs to be fixed in a way such that these tests are independent of timezone.

Reply:
@mseaton indeed there was an issue with timezone. It has corrected. You can check it now

Reply:
Merged!  Thanks for the patience and continued effort - it was a long road but we got there!

Reply:
Thanks for thorough code review and tips that made the code nicer and more effective